go/kbfs/libkbfs/folder_update_prepper.go

```diff
@@ -376,6 +376,13 @@ func (fup *folderUpdatePrepper) prepUpdateForPath(
 			for _, unref := range unrefs {
 				md.AddUnrefBlock(unref)
 			}
+			// Fetch the current block again, since `setEntry` might
+			// not modify the original `currBlock`, but some
+			// re-assembled version if the disk cache is in use.
+			currBlock, err = dbm.getBlock(ctx, currDD.rootBlockPointer())
```

Comment:
why does getting it again prevent us from modifying a re-assembled version? Because if it's been queried recently it will live in the memory cache instead?

Reply:
The comment is about the earlier call to `setEntry()`.  That code adds a new entry to the directory, but it doesn't use `currBlock`.  Instead, it `Get`s the block from the `dirBlockMap`, which if it's disk-based, ends up constructing a new instance of the block, and that instance is the one that gets modified and written back to disk.  So this `getBlock` call here is to read the newly-modified block from disk and replace `currBlock` with it.